src/eval.rs

```diff
@@ -183,11 +223,44 @@ impl Expr {
                     }
                 }))
             }
+            ExprSource::Map { .. } => Err(EvalError::Internal(InternalError::Unexpected(
+                "eval_uncached ExprSource::Map should be unreachable, ".to_string()
+                    + "since the Expr::value should be initialized at creation",
+            ))),
+            ExprSource::MapAttr { value, .. } => value.as_ref()?.eval(),
+            ExprSource::Dynamic { inner } => inner.as_ref()?.eval(),
+            ExprSource::Ident { name } => self
+                .scope
+                .get(name)
+                // We don't have everything implemented yet, so assume we're at fault
+                .ok_or(EvalError::Internal(InternalError::Unimplemented(format!(
+                    "not found in scope: {}",
```

Comment:
Is the inner error somehow exposed to the user? If so, it should be made explicit (unless the `Unimplemented` is already made clear as well).

Reply:
`InternalError` errors are silent. I agree that it should be more explicit; I've pushed https://github.com/nix-community/rnix-lsp/pull/55/commits/597ee5c4fa5fc72de67bbb54d48cf11a33f37495 to improve the comments